### ISSUE TITLE
Unit Test Bias Estimation Failure

### ISSUE THREAD

**Issue by mfacchinelli** [Tudat/tudat#423]
Hi Dominic, 

As I mentioned, the `unitTestBiasEstimation.cpp` starts failing after [this](https://github.com/mfacchinelli/tudat/commit/0c1aed80052db56588a472aa3b743eabbf1bf559) commit and is related to the new constructor for the Runge-Kutta variable step size integration settings. As it turns out, the error it gives in this commit is different from the one I told you about. It is related to memory access violation. I tried replacing the integrator settings such that the input template is a matrix, as in:
```
    // Create integrator settings
    boost::shared_ptr< IntegratorSettings< TimeType > > integratorSettings =
            boost::make_shared< RungeKuttaVariableStepSizeSettings< TimeType,
            Eigen::Matrix< StateScalarType, Eigen::Dynamic, Eigen::Dynamic > > >
            ( rungeKuttaVariableStepSize, TimeType( initialEphemerisTime ), 120.0,
              RungeKuttaCoefficients::CoefficientSets::rungeKuttaFehlberg78,
              120.0, 120.0, 1.0, 1.0 );
```
but it still gives the same error. What I know for certain is that this is the first commit where any error appears. 

If you replace `createNumericalIntegrator.h` with its newest version (which you can find [here](https://github.com/mfacchinelli/tudat/blob/master/Tudat/Mathematics/NumericalIntegrators/createNumericalIntegrator.h)), the `NaN` error comes out (when `Eigen::Matrix< StateScalarType, Eigen::Dynamic, Eigen::Dynamic >` is the second template argument). I guess that the earlier version had some other issue as well.

Thank you and kind regards, 

Michele

**Comment by DominicDirkx** [comment]
Hi Michele,

I'm a bit confused. You mention that there are two different errors, one in the commit you link to, and another one (in the latest commit?). Do you think these are the same errors? Is the first one solved in the latest commit, but is there a new one?

Dominic

**Comment by DominicDirkx** [comment]
HI Michele, I've found what's causing the crash in the commit you mention: no integrator is created, as the template arguments don't match (VectorXd vs MatrixXd). An error was not thrown due to a missing 'throw' command. See commit here:

**Comment by DominicDirkx** [comment]
Hi Michele,

I've fixed the crash, see:

https://github.com/DominicDirkx/tudat/commit/60c28b7a844c35f153f5c6f7ed63ef9ee3dcaddc

It still returns NaN, though. I realized that the RungeKuttaVariableStepSizeSettings should really not have a StateType template argument (I'd be happy to explain/discuss why in person, doing it here is not efficient). Could you remove it in the final version of teh code (and the new constructor). I have an idea for how to let users access the new functionality of per-element tolerances, but this can be added later,

I'll try to find the time to find the reason for teh NaN result

Best,

Dominic

**Comment by mfacchinelli** [comment]
Hi Dominic, 

Yes, that bug I had found as well, but I forgot it was the cause of that error. The main problem with the bias estimation test was indeed the `NaN`. 

Ok, I will revert the changes and on Friday you can explain the new method.

Thank you!

Michele

**Comment by DominicDirkx** [comment]
Sounds good! Because I was curious, I tested the code without the extra template parameter is the settings class, and the estimation runs as expected! My best guess is that something was going wrong with the newStepFunction.

If you could push the small modifications (removing the second template argument), I'll merge the new code and documentation.

**Comment by mfacchinelli** [comment]
Hi Dominic, 

I tried removing only the step size function, but the same error popped up. So I guess it's not (only) related to that. I have removed the extra template and indeed the test is passed successfully. 

I will now test the rest of the functionalities to make sure that everything else works fine. Once I am done I will create the pull request.

Have a nice day and see you Friday, 

Michele
